src/pages/reviewed-preprints/[msid]/[...tab].page.tsx

```diff
@@ -27,24 +28,38 @@ const getPublishedDate = (events: TimelineEvent[]): string | undefined => {
   return undefined;
 };
 
-export const Page = (props: PageProps): JSX.Element => (
-  <>
-  <Head>
-    <title>{contentToText(props.metaData.title)}</title>
-    <meta name="citation_title" content={contentToText(props.metaData.title)}/>
-    <meta name="citation_publisher" content="eLife Sciences Publications Limited"/>
-    <meta name="citation_doi" content={props.metaData.doi}/>
-    <meta name="citation_publication_date" content={getPublishedDate(props.status.timeline)}/>
-    <meta name="citation_pdf_url" content={props.metaData.pdfUrl}/>
-    <meta name="citation_fulltext_html_url" content={`https://elifesciences.org/reviewed-preprints/${props.metaData.msid}`}/>
-    <meta name="citation_language" content="text/html"/>
-    { props.metaData.authors.map((author, index) => <meta key={index} name="citation_author" content={`${author.givenNames?.join(' ')} ${author.familyNames?.join(' ')}`}/>)}
-  </Head>
-  <ArticlePage metaData={props.metaData} msidWithVersion={props.msidWithVersion} status={props.status} activeTab="fulltext">
-    <ArticleFullTextTab content={props.content} metaData={props.metaData} peerReview={props.peerReview}></ArticleFullTextTab>
-  </ArticlePage>
-  </>
-);
+export const Page = (props: PageProps): JSX.Element => {
+  const tabs: { [key: string]: () => JSX.Element } = {
+    fulltext: () => <ArticleFullTextTab content={props.content} metaData={props.metaData} peerReview={props.peerReview}></ArticleFullTextTab>,
+    figures: () => <ArticleFiguresTab content={props.content}></ArticleFiguresTab>,
+    reviews: () => <ArticleReviewsTab peerReview={props.peerReview}></ArticleReviewsTab>,
+    pdf: () => (<>
+        <ArticleFullTextTab content={props.content} metaData={props.metaData} peerReview={props.peerReview}></ArticleFullTextTab>
+        <ArticleReviewsTab peerReview={props.peerReview}></ArticleReviewsTab>
+      </>),
+  };
+  const router = useRouter();
+  const tabName = (router.query.tab?.[0] && tabs[router.query.tab[0]] !== undefined) ? router.query.tab[0] : 'fulltext';
```

Comment:
I'm wondering if this line and the line below would be better served as memos that change dependent on router.query.tab so they don't recalculate upon every render on the client (unless that's not how Next handles it?)

Reply:
Adding a console log to track when it's trigger, it looks to be only on page load that that page component is called.

Having said that - if that was to change there's no harm in adding the `useMemo` and it seems to work as we expect around `router.query.tab` (or at least triggers as often as it ever did). Added! 👍